### PR TITLE
Add filter command

### LINKED ISSUE #80: Implement Filter by NextLesson Date Feature

### PR DESCRIPTION
**Pull request by nicolaskjh**
Closes #80

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/FilterDateCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        assertParseFailure(parser, "15-04-2025", MESSAGE_INVALID_DATE);
+
+        // invalid day
+        assertParseFailure(parser, "0/4/2025", MESSAGE_INVALID_DATE);
+
+        // invalid month
+        assertParseFailure(parser, "15/14/2025", MESSAGE_INVALID_DATE);
+
+        // invalid year
+        assertParseFailure(parser, "15/4/202", MESSAGE_INVALID_DATE);
```
Comment (bipplane) [inline src/test/java/seedu/address/logic/parser/FilterDateCommandParserTest.java:72]: great testing of edge cases!

File: src/main/java/seedu/address/logic/parser/FilterDateCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+            if (filterNextLessonDate.isBefore(CURRENT_DATE)) {
+                throw new ParseException(MESSAGE_INVALID_PAST_DATE, new IllegalArgumentException());
+            }
+
+            return new NextLessonEqualsDatePredicate(filterNextLessonDate);
+        } catch (DateTimeException e) {
+            throw new ParseException(MESSAGE_INVALID_DATE, e);
+        }
+    }
```
Comment (bipplane) [inline src/main/java/seedu/address/logic/parser/FilterDateCommandParser.java:63]: nice use of a helper method!

**Review by bipplane — APPROVED** [review]
Overall LGTM! This would be a great feature to add into our TutorRec application!

**Review by flljy940 — APPROVED** [review]
LGTM!

**Review by izniy — APPROVED** [review]
LGTM. Filter is a valuable command for our TutorRec application!
